lib/executor.js

```diff
@@ -305,15 +305,20 @@ function runScripts(app, list, callback) {
 
   async.eachSeries(functions, function(f, done) {
     debug('Running script %s', f.path);
-    if (f.func.length >= 2) {
-      debug('Starting async function %s', f.path);
-      f.func(app, function(err) {
-        debug('Async function finished %s', f.path);
-        done(err);
-      });
-    } else {
-      debug('Starting sync function %s', f.path);
-      f.func(app);
+    var cb = function(err) {
+      debug('Async function finished %s', f.path);
+      done(err);
+    };
+    var result, error;
+    try {
+      result = f.func(app, cb);
+    } catch (err) {
+      debug('Sync function failed %s', f.path, err);
+      error = err;
```

Comment:
AFAICT, `error` is not used. Can we remove it please?

Also when the function exported by a boot script fails, the error should be reported back/up to the caller of `boot(app)`. I think you should call `return done(err)` here.